linetools/isgm/utils.py

```diff
@@ -362,3 +366,138 @@ def synthesize_components(components, zcomp=None, vbuff=0*u.km/u.s):
+    else:
+        chunk1 = np.array(chunk1)  # not a quantity
+
+    # this may be redundant but cleaner code
+    if isinstance(chunk2, Quantity):
+        unit2 = chunk2.unit
+        chunk2 = np.array(chunk2.value)
+        if not isinstance(chunk1, Quantity):
+            raise ValueError('chunk1 must be Quantity because chunk2 is!')
+    else:
+        chunk2 = np.array(chunk2)  # not a quantity
+    # here we have chunks as values (i.e no units)
+
+    # make sure the chunks are sorted
+    cond1 = np.sort(chunk1) != chunk1
+    cond2 = np.sort(chunk2) != chunk2
+    if (np.sum(cond1) > 0) or (np.sum(cond2) > 0):
+        raise ValueError('chunks must be sorted!')
+
+    # figure out the lowest of the chunks
+    # and sort them such that chunk1 is by definition
+    # the one with the lowest limit
+    if np.min(chunk1) <= np.min(chunk2):
+        pass
+    else: # invert them if necessary
+        aux = chunk1
+        chunk1 = chunk2
+        chunk2 = aux
+
+    # now the chunk1 is the one with the lowest value
+    # then, the only way they do not overlap is:
+    if np.min(chunk2) > np.max(chunk1):
+        return False
+    else:  # overlap exists
+        return True
+
+def overlapping_components(comp1, comp2, tol=0.2*u.arcsec):
+    """Whether two components overlap in wavelength
+    space and (ra,dec) sky position. This is useful to identify
+    components that may need to be fit together.
+
+    Parameters
+    ----------
+    comp1 : AbsComponent
+        A given AbsComponent object
+    comp2 : AbsComponent
+        A given AbsComponent object
+    tol : Quantity, optional
+        Tolerance for checking whether the two components are
+        in the same sky region. Default is 0.2*u.arcsec
+
+    Returns
+    -------
+    Boolean : True if there is overlapping wavelength range
+    and radec coordinates, otherwise False.
+    """
+
+    if not isinstance(comp1, AbsComponent):
+        raise ValueError('comp1 must be AbsComponent object.')
+    if not isinstance(comp2, AbsComponent):
+        raise ValueError('comp1 must be AbsComponent object.')
+
+    # Check whether they are in the same sky region
+    if comp1.coord.separation(comp2.coord) > tol:
+        return False
+
+    # Define wavelength chunks, appending them from each absline
+    # These will correpond to (wvmin, wvmax) tuples for each absline
+    # in the respective component
+    wobs_chunks_1 = []
+    wobs_chunks_2 = []
+    for absline in comp1._abslines:
+        cond = absline.analy['vlim'] != init_analy['vlim']  # default value?
+        if np.sum(cond) > 0:  # i.e. absline vlim not equal than the default
+            dzlim = give_dz(absline.analy['vlim'], comp1.zcomp)
+        else:  # use the vlim from component otherwise
+            dzlim = give_dz(comp1.vlim, comp1.zcomp)
+        wrest_aux = absline.wrest.to('AA').value  # in AA
+        wobs_aux = wrest_aux * (1 + comp1.zcomp + dzlim)
+        wobs_chunks_1 += [tuple(wobs_aux)]
```

Comment:
I kind of like the idea of having the equivalent method for individual abslines and will probably do it that way.

The idea is to identify components whose abslines share the same part of a given spectra, so the fit must be done simultaneously. It is written in the main docstring of the method, but may need extra explanation.